File: code/Server/src/model/cubeScanner/cubeScanner.ts
import db from "../../db/db";

const updateIpAddress = async (pickupPointPosition: number, ipAddress: string) => {
  try {
    const pickupPoint = await db.pickup_point.findFirst({
      where: {
        position: pickupPointPosition,
      },
    });

    if (pickupPoint == null) {
      console.log("no pickup point found in position " + pickupPointPosition);
      return;
    }

    await db.cube_scanner.updateMany({
      data: {
        ip_address: ipAddress,
      },
      where: {
        id_pickup_point: pickupPoint.id,
      },
    });
  } catch (error) {
    console.log(error);
  }
};

const cubeScanner = {
  updateIpAddress,
};

export default cubeScanner;

File: code/Mobile Application/lib/data/model/cube_scanner.dart
import 'package:sm_iot_lab/data/model/pickup_point.dart';

class CubeScanner {
  int id;
  String? ipAddress;
  int position;
  PickupPoint? pickupPoint;

  CubeScanner({
    required this.id,
    required this.ipAddress,
    required this.position,
    this.pickupPoint,
  });

  static CubeScanner fromMap(Map<String, dynamic> data) {
    return CubeScanner(
      id: data['id'],
      ipAddress: data['ip_address'],
      position: data['pickup_point']["position"],
    );
  }
}
